clang/test/Sema/asm.c

```diff
@@ -365,3 +365,9 @@ void test19(long long x)
   // FIXME: This case should be supported by codegen, but it fails now.
   asm ("" : "=rm" (x): "0" (e)); // expected-error {{unsupported inline asm: input with type 'st_size128' (aka 'struct _st_size128') matching output with type 'long long'}}
 }
+
+// GH118892
+void test20(char x) {
+  double value;
+  asm ("fabs" : "=t" (value): "0" (x)); // expected-error {{unsupported inline asm: input with type 'char' matching output with type 'double'}}
+}
```

Comment:
Test both directions

Reply:
Not sure about vectors (is it `std::vector`)? I'll add test against structs and floats.

Reply:
No, I mean like int2 `typedef int int2 __attribute__((ext_vector_type(2)))`

Reply:
Added.